standards/ICRC-3/README.md

```diff
@@ -0,0 +1,206 @@
+# `ICRC-3`: Transaction Log
+
+| Status |
+|:------:|
+| Draft  |
+
+`ICRC-3` is a standard for accessing the transaction log of a Ledger on the [Internet Computer](https://internetcomputer.org).
+
+`ICRC-3` specifies:
+1. A generic format for sharing the transaction log without information loss
+2. A mechanism to verify the transaction log on the client side to allow downloading the transaction log via query calls
+3. A way for new standards to define new transaction types compatible with ICRC-3
+
+## Transaction Log
+
+The transaction log is a list of transactions where each transaction contains the hash of its parent (`phash`). The parent of a transaction `i` is transaction `i-1` for `i>0` and `null` for `i=0`.
+
+```
+   ┌──────────────────────┐          ┌──────────────────────┐
+   |         Tx i         |          |         Tx i+1       |
+   ├──────────────────────┤          ├──────────────────────┤
+◄──| phash = hash(Tx i-1) |◄─────────| phash = hash(Tx i)   |
+   | ...                  |          | ...                  |
+   └──────────────────────┘          └──────────────────────┘
+
+```
+
+## Value
+
+The [candid](https://github.com/dfinity/candid) format supports sharing information even when the client and the server involved do not have the same schema (see the [Upgrading and subtyping](https://github.com/dfinity/candid/blob/master/spec/Candid.md#upgrading-and-subtyping) section of the candid spec). While this mechanism allows to evolve services and clients
+independently without breaking them, it also means that a client may not receive all the information that the server is sending, e.g. in case the client schema lacks some fields that the server schema has.
+
+This loss of information is not an option for `ICRC-3`. The client must receive the same exact data the server sent. For this reason, `ICRC-3` introduces the `Value` type which never changes: 
+
+```
+type Value = variant { 
+    Blob : blob; 
+    Text : text; 
+    Nat : nat; // do we need this or can we just use Int?
+    Int : int;
+    Array : vec Value; 
+    Map : vec record { text; Value }; 
+};
+```
+
+Servers must serve the transaction log as list of `Value` where each `Value` represent a single transaction in the transaction log.
+
+## Value Hash
+
+`ICRC-3` specifies a standard hash function over `Value`.
+
+This hash function should be used by Ledgers to calculate the hash of the parent of a transaction and by clients to verify the downloaded transaction log.
+
+The hash function is the [representation-independent hashing of structured data](https://internetcomputer.org/docs/current/references/ic-interface-spec#hash-of-map) used by the IC:
+- the hash of a `Blob` is the hash of the bytes themselves
+- the hash of a `Text` is the hash of the bytes representing the text
+- the hash of a `Nat` is the [`leb128`](https://en.wikipedia.org/wiki/LEB128#Unsigned_LEB128) encoding of the number
```

Comment:
I think it should be the hash. Otherwise there's no bound on the hash of a Value and that looks undesirable and a somewhat inconsistent. Having said that, we'd never have to hash a raw Nat or Int; these would always be part of some outer structure which would be compressed.

Reply:
Fixed with 0e4b103.